internal/types/types.go

```diff
@@ -0,0 +1,3 @@
+package types
+
+type Unit struct{}
```

Comment:
That said, `types.Unit` is way more descriptive for what it's worth. From the DX perspective, I want there to be no semantic overhead on the usage of the API. This is a tradeoff worth making

Reply:
is is though?

i won't block merging on this, as long as it gets wrapped up inside `femto.EmptyResponce` so it's not used everywhere. but it's not descriptive at all (relies on knowing type-theory stuff), and adding an entire package so that we can have a unit type **with a longer name** than the completely descriptive `struct{}` is super terable

Reply:
I think it's not a bad bet to assume that the programmer using this API (us) or any maintainer in the future would be familiar with the idea of unit.